### ISSUE TITLE
this work tho 15" XPS model?

### ISSUE THREAD

**Issue by SerkanDayicik** [syscl/XPS9350-macOS#40]
hello there, i want to buy XPS 15" Model on turkey what best model for this can u share any link on amazon ?

**Comment by R1MnTnA** [comment]
Do not post questions of this kind here. Please visit some forums online. Like tonymacx86.com or insanelymac.com

Thank you

**Comment by syscl** [comment]
@SerkanDayicik This can in theory transform to XPS 13 9360 and XPS 15 9550. 

If you pay attention to my previous M3800 and Asus-FX50J(X/K) project, you will soon found this project can be easily deployed on other platform. XPS 15 and XPS 13 skylake has very little difference, only changed some ACPI table patches and config.plist setting, audio resources, then it will work.

I've asked @wmchris if he is interested in porting this project to XPS 15 9550, but he may take some time to port it(I guess). One great XPS 15 9550 project can be referred to @darkhandz [here](https://github.com/darkhandz) . 

Wish this help,
syscl

**Comment by shawnhwei** [comment]
@syscl Is there any advantage to porting this project over using darkhandz existing build?

**Comment by syscl** [comment]
@shawnhwei Lets list some advantages by using deploy:

- Dynamic patch ACPI tables per time once your BIOS settings get changed, or your BIOS has been updated/downgraded. This prevents exhausted ACPI table patches per time, it's pretty time wasted and will very likely do something wrong by manually patching ACPI tables.
- Use original ACPI tables from you own laptops
- USB sleep eject problem get fixed while calling deploy
- USB Wi-Fi cause sleep panic get fixed while using deploy
- Auto update all kext/drivers per call 
- Compatible to previous config, and will try to update config.plist to latest version
- Convenience, finish the whole post install by simply typing deploy
- Auto patch Recovery HD for high resolution laptop support(QHD+/4K) and prevent multiple patch per call
- Optimized X86PlatformPluginInjector install by deploy automatically
- ...
- ...

syscl

**Comment by shawnhwei** [comment]
@syscl In that case I am willing to attempt to port it if it is as straightforward as you said previously.

**Comment by wmchris** [comment]
i'm using darkhands and i dont have any of the issues above. Thats why i didnt port it.

**Comment by shawnhwei** [comment]
@wmchris They are also working for me, aside from the nvme corruption issue mentioned in this repo as well. You haven't made any changes to darkhandz's setup?

**Comment by wmchris** [comment]
We both did many changes, and we implemented them both. So they're pretty much the same, but i also added some different approaches.
NVMe corruption is fixable on many machines:
http://www.insanelymac.com/forum/topic/312803-patch-for-using-nvme-under-macos-sierra-is-ready/?p=2377304
if your machine is not compatible: you'll have the issue on darkhands as well as on the deploy method.

**Comment by syscl** [comment]
@wmchris If you've experienced the update from 10.11.1/10.11.2/10.11.3 to 10.11.4 or above, you will soon know why deploy is a good solution and can be proved to be robust. A lot people use old AppleHDA while deploy just generate AppleHDA_Injector from your current latest AppleHDA, which has not a single issue while updating. 

You're just luck enough to use old ACPI tables and some old stuff. And I don't know what procedure should need a redo, what you need to do is just deploy execute, after executing within one minute, everything is done. 

Manually patch ACPI tables should be done in experiment stage, once you finish all require ACPI patches procedures, you shouldn't do it again and again, time wasted and very likely to do things wrong.

syscl

**Comment by wmchris** [comment]
@syscl of course i know, i've got thru all of this and i used my own deploy system on 10.11, but with the recent discoveries (where you also helped a lot ;) ) it's simply not needed anymore.

darkhands system uses the live ACPI tables, not old ones, and live patches them. this is pretty much the best you can do - always using the newest.

**Comment by syscl** [comment]
@wmchris I was those who tried to implement the so-called tiny SSDT method very early ago. But, from the experiment for Dell Precision M3800, not all SSDT tables work as expect, one of the case is:
```
ALSD -> ALS0
```

And also, why should not use original tables then patch it directly? It's most efficient and straightforward to patch things directly if there exist already. I also want to port this project for XPS 15 9550 as well, but seems nobody tend to give me feedback thus I can do nothing..

AppleALC is good for vanilla AppleHDA, and I am those person who implement AppleALC in my Dell Precision M3800 project when AppleALC was at the initial stage. But, the problem is, AppleALC does a lot of operations that will very likely cause kernel panic(e.g. disable some interrupts in versions which is very very very dangerous). It's good and I like AppleALC project, but for stable and follow ```less is more``` policy, I will let deploy to generate AppleHDA_Injector that is most suitable for the current system.

syscl

**Comment by wmchris** [comment]
yes, thats true, the AppleHDA injector is much better than VoodooHDA or AppleALC. I prefer your method and i also added that in my repo (of course with credit to you :-) ). However on the 9550 live SSDT patching works flawless. It's not the perfect way for every machine, but for the 9550 its pretty good. 

I tried your deploy algorithm in the past and i had no benefits from it, thats why i didnt finish the port. :/

**Comment by syscl** [comment]
I will try to read from your thread then port to it someday, though I didn't have any XPS 15 9550 model, but I saw the major difference still lies in:
- ACPI patches(easy to handle), including disable discrete graphics card like M3800's deploy project 
- OsxAptioFixDrv instead of OsxAptioFix2Drv
- ig-platform-id
- AppleHDA_Injector(already finished by you)
- Please let me know if there has something I've missed

Thank you,
syscl

**Comment by wmchris** [comment]
you need:
4k sector fix
some patches in config.plist like minstolen
4k monitor retina perl command

the rest is optional.

**Comment by shawnhwei** [comment]
@syscl so it seems like there is still motivation to pursue this. Maybe I'll go ahead and start a fork and PR back to you when you have time.

**Comment by syscl** [comment]
@shawnhwei Thank you!

syscl
